### ISSUE TITLE
CSV download of the feedback responses should take the groupBy dropdown selection in to consideration

### ISSUE THREAD

**Issue by rclakmal** [TEAMMATES/teammates#5630]
Currently CSV download only supports a default groupBy option. 
![groupby](https://cloud.githubusercontent.com/assets/794011/16062256/405ccd7a-3292-11e6-9cda-63e25f6e214b.png)

**Comment by rclakmal** [comment]
@damithc This is bit ambiguous and complex than I expected. Although the group-by feature makes sense in panel view, does it really applies for a CSV download ?

or rather should we use this selected value (example` G>R>Q`, `R>G>Q`) to decide how we should sort the CSV responses ? For an example currently, it is sorted by default to G>R>Q by calling following line

       ```
 // sort responses by giver > recipient > qnNumber
        Collections.sort(results.responses,
                results.compareByGiverRecipientQuestion);
```

**Comment by damithc** [comment]
Yes, group-by will be harder to do. Shall we finish the 'Section' field first?. i.e. include only the section shown. Meanwhile, let me think about the 'group by' option.

**Comment by rclakmal** [comment]
@damithc Section choice already been refereed by CSV download. We should close #5628.

**Comment by damithc** [comment]
I see. In that case, work on that while we figure out what to do with this one. I'll get back with my thoughts later today.

**Comment by damithc** [comment]
Now that I think about it, a csv formatted data in various groupings (G>R>Q etc.) may not be that useful. Perhaps we should provide a 'print' option so that users can download the results as they are seen in the webpage? What do you think @kanghj @rclakmal ?
On a different front, we can provide more filtering options so that the user can view specific teams/students/questions on the webpage and also download those filtered data.

**Comment by rclakmal** [comment]
@damithc By print, do you mean export the view as images and probably present it as a 'pdf' ?

I believe we already have an item in our proposal tracker to introduce more filters to session view like in student view.

**Comment by damithc** [comment]
Not really export to pdf, but rather, a 'printer friendly' button that formats the page for printing. Users can print as pdf if they want. Investigate how do others usually facilitate a printer friendly view.

**Comment by rclakmal** [comment]
Closed with #5779
